src/api/venue.ts

```diff
@@ -34,3 +47,144 @@ export const setVenueLiveStatus = async ({
+  venueId: string
+): Promise<WithId<AnyVenue> | undefined> =>
+  getVenueRef(venueId)
+    .withConverter(anyVenueWithIdConverter)
+    .get()
+    .then((docSnapshot) => docSnapshot.data());
+
+export const fetchDirectChildVenues = async (
+  venueIdOrIds: string | string[]
+): Promise<WithId<AnyVenue>[]> => {
+  const venueIds: string[] =
+    typeof venueIdOrIds === "string" ? [venueIdOrIds] : venueIdOrIds;
+
+  if (venueIds.length <= 0) return [];
+
+  return Promise.all(
+    chunk(venueIds, FIRESTORE_QUERY_IN_ARRAY_MAX_ITEMS).map(
+      async (venueIdsChunk: string[]) => {
+        const childVenuesSnapshot = await getVenueCollectionRef()
+          .where("parentId", "in", venueIdsChunk)
+          .withConverter(anyVenueWithIdConverter)
+          .get();
+
+        return childVenuesSnapshot.docs
+          .filter((docSnapshot) => docSnapshot.exists)
+          .map((docSnapshot) => docSnapshot.data());
+      }
+    )
+  ).then((result) => result.flat());
+};
+
+export interface FetchDescendantVenuesOptions {
+  maxDepth?: number;
+}
+
+export const fetchDescendantVenues = async (
+  venueIdOrIds: string | string[],
+  options?: FetchDescendantVenuesOptions
+): Promise<WithId<AnyVenue>[]> => {
+  const venueIds: string[] =
+    typeof venueIdOrIds === "string" ? [venueIdOrIds] : venueIdOrIds;
+
+  const { maxDepth } = options ?? {};
+
+  const directChildVenues: WithId<AnyVenue>[] = await fetchDirectChildVenues(
+    venueIds
+  );
+
+  if (maxDepth && maxDepth <= 0 && directChildVenues.length > 0)
+    throw new Error(
+      "Maximum depth reached before fetching all descendant venues."
+    );
+
+  const descendantVenues: WithId<AnyVenue>[] = await fetchDescendantVenues(
+    directChildVenues.map((venue) => venue.id),
+    {
+      ...options,
+      maxDepth: maxDepth ? maxDepth - 1 : undefined,
+    }
+  );
+
+  return [...directChildVenues, ...descendantVenues];
+};
+
+export const fetchRelatedVenues = async (
+  venueId: string
+): Promise<WithId<AnyVenue>[]> => {
+  const { sovereignVenue } = await fetchSovereignVenue(venueId);
+
+  const descendantVenues = await fetchDescendantVenues(sovereignVenue.id);
+
+  return [sovereignVenue, ...descendantVenues];
+};
+
+/**
+ * Convert Venue objects between the app/firestore formats (@debt:, including validation).
+ */
+export const anyVenueWithIdConverter: firebase.firestore.FirestoreDataConverter<
+  WithId<AnyVenue>
+> = {
+  toFirestore: (
+    anyVenue: WithId<AnyVenue>
+  ): firebase.firestore.DocumentData => {
+    // @debt Properly check/validate this data
+    //   return AnyVenueSchema.validateSync(anyVenue);
+
+    return anyVenue;
+  },
+
+  fromFirestore: (
+    snapshot: firebase.firestore.QueryDocumentSnapshot
+  ): WithId<AnyVenue> => {
+    // @debt Properly check/validate this data rather than using 'as'
+    //   return withId(AnyVenueSchema.validateSync(snapshot.data(), snapshot.id);
+
+    return withId(snapshot.data() as AnyVenue, snapshot.id);
+  },
```

Comment:
[debt] Ideally we would implement these validators as part of the converter as well, but given the state of our Venue types/validations/etc at the moment, I think that would be more pain than we have time for in this PR. As is, this is probably no worse off than any of the rest of our codebase at the moment (which also doesn't validate data when reading back out of the database)